azurerm/validators.go

```diff
@@ -3,9 +3,31 @@ package azurerm
 import (
 	"fmt"
 
+	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/satori/uuid"
 )
 
+// validateIntInSlice returns a SchemaValidateFunc which tests if the provided value
+// is of type int and matches the value of an element in the valid slice
+func validateIntInSlice(valid []int) schema.SchemaValidateFunc {
+	return func(i interface{}, k string) (s []string, es []error) {
+		v, ok := i.(int)
+		if !ok {
+			es = append(es, fmt.Errorf("expected type of %s to be int", k))
```

Comment:
Nitpick: It might be better to use `%q` as quotes can make it clearer it's a field name, not just another word. e.g. as a user if I saw `expected type of key to be int` I'd be tempted to think "which key?".

Reply:
Good point. Updated.